packages/editor/src/components/post-author/enhanced-post-author.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * External dependencies
+ */
+import { debounce } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import ComboboxControl from '../../../../components/build/combobox-control/';
+
+function EnhancedPostAuthor( { postAuthor, id, authors, onUpdateAuthor } ) {
+	const currentPostAuthor = postAuthor[ 0 ];
+	let initialAuthors = [];
+	authors.forEach( ( author, i ) => {
+		initialAuthors = [
+			...initialAuthors,
+			{
+				key: i,
+				name: author.name,
+				value: author.name,
+				id: author.id,
+			},
+		];
+	} );
+
+	// Ensure the current author is included in the initial dropdown list.
+	let foundAuthor = initialAuthors.findIndex(
+		( author ) => currentPostAuthor && currentPostAuthor.id === author.id
+	);
+
+	if ( currentPostAuthor && foundAuthor < 0 ) {
```

Comment:
What happens when there is no `currentPostAuthor`?

Reply:
There is always a post author, for new posts this is the current user.

Reply:
Then why do you check if it's truthy in the conditional?